sources/api/netdog/src/networkd_status.rs

```diff
@@ -0,0 +1,304 @@
+//! The networkd_status module contains definitions and functions for tracking network status from networkd
+//!
+use crate::interface_id::InterfaceName;
+#[cfg(net_backend = "systemd-networkd")]
+use crate::NETWORKCTL;
+use serde::{Deserialize, Deserializer};
+#[cfg(net_backend = "systemd-networkd")]
+use snafu::{ensure, ResultExt};
+#[cfg(net_backend = "systemd-networkd")]
+use std::cmp::Ordering;
+use std::convert::TryInto;
+use std::net::IpAddr;
+#[cfg(net_backend = "systemd-networkd")]
+use std::process::Command;
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[allow(dead_code)]
+pub(crate) struct NetworkdDnsConfig {
+    #[serde(rename = "Family")]
+    family: u8,
+    #[serde(rename = "Address", deserialize_with = "ipaddr_from_vec_de")]
+    pub(crate) address: IpAddr,
+    #[serde(rename = "ConfigSource")]
+    config_source: String,
+    #[serde(rename = "ConfigProvider", deserialize_with = "ipaddr_from_vec_de")]
+    config_provider: IpAddr,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[allow(dead_code)]
+pub(crate) struct SearchDomain {
+    #[serde(rename = "Domain")]
+    pub(crate) domain: String,
+    #[serde(rename = "ConfigSource")]
+    config_source: String,
+    #[serde(rename = "ConfigProvider", deserialize_with = "ipaddr_from_vec_de")]
+    config_provider: IpAddr,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct NetworkctlIpAddr {}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+#[allow(dead_code)]
+pub(crate) struct NetworkdStatus {
```

Comment:
Naming nit:  This struct contains the status for a particular interface, not Netword.  Perhaps something like `InterfaceStatus`?

Style nit:  Since this is the "top level" struct, that contains everything else I'd put it at the top of the file.

Reply:
I didn't want to be overly specific but yes, the true name would be something akin to `NetworkdInterfaceStatus` which isn't too bad so I can update it to be more correct. But it  *is* specific to networkd's "view" of this interface.